dvc/repo/reproduce.py

```diff
@@ -32,6 +32,23 @@ def _reproduce_stage(stages, node, **kwargs):
     return [stage]
 
 
+def _get_active_graph(G):
+    import networkx as nx
+
+    active = G.copy()
+    stages = nx.get_node_attributes(G, "stage")
+    for node in G.nodes():
```

Comment:
As I can see you don't need to call `.nodes()`, it's possible to iterate graph directly and there is no need to prepare attrs you may use `stage = G.nodes[node]['stage']`. Or iterate `G.nodes.items()`

BTW, why don't we add stages to graph directly? Then it would be simply:

```python
for stage in G:
    if not stage.locked:
        continue
    # ...
```

Reply:
Addressed the `nodes()` issue.

> BTW, why don't we add stages to graph directly? Then it would be simply:

Mostly a legacy I think. Might take a look at some point.